app/controllers/notes_controller.rb

```diff
@@ -0,0 +1,55 @@
+class NotesController < ApplicationController
+  before_action :set_note, only: %i[edit update destroy]
+  before_action :set_user_challenge, only: %i[new create edit]
+
+  def new
+    @note = Note.new
+  end
+
+  def create
+    @note = current_user.notes.build(note_params)
+    @note.user_challenge = @user_challenge
+    if @note.save
+      respond_to do |format|
+        format.html { redirect_to user_challenge_path(@user_challenge) }
+        format.turbo_stream
+      end
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @note.update(note_params)
+      redirect_to user_challenge_path(@note.user_challenge)
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @note.destroy
+
+    respond_to do |format|
+      format.html { redirect_to user_challenge_path(@note.user_challenge), status: :see_other }
+      format.turbo_stream
+    end
+  end
+
+  private
+
+  def note_params
+    params.require(:note).permit(:note)
```

Comment:
I'm not a fan that the model is called note and the column is also note; if it works, it's fine, but maybe message would have been easier to differentiate

Reply:
Basically, the note column already existed in the 'user_challenge' model but we decided to have a separate model 'note' since a user can create many notes on each user_challenge.
I didn't want to break any code so just left the 'note' column for now so it's redundant but once we clear the code, we can remove the 'note' column from the 'user_challenge' model.